From 30e1afe60820c18c89cd10e9dcb7208ad5676d1d Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 19 Jan 2023 15:06:35 -0800
Subject: [PATCH] xmldsigverify: Include xmlsec/parser.h

this ensures that xmlSecParserSetDefaultOptions definition is made
available

Fixes
| xmldsigverify.c:275:5: error: call to undeclared function 'xmlSecParserSetDefaultOptions'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
|     xmlSecParserSetDefaultOptions(XML_PARSE_NOENT | XML_PARSE_NOCDATA |
|     ^

Upstream-Status: Submitted [https://github.com/lsh123/xmlsec/pull/463]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 examples/xmldsigverify.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/examples/xmldsigverify.c b/examples/xmldsigverify.c
index ed538c70..ff68120e 100644
--- a/examples/xmldsigverify.c
+++ b/examples/xmldsigverify.c
@@ -25,6 +25,7 @@
 #include <xmlsec/xmldsig.h>
 #include <xmlsec/crypto.h>
 
+#include <xmlsec/parser.h>
 /* #define XMLDSIGVERIFY_DEFAULT_TRUSTED_CERTS_FOLDER   "/etc/httpd/conf/ssl.crt" */
 #define XMLDSIGVERIFY_DEFAULT_TRUSTED_CERTS_FOLDER      "/var/www/cgi-bin/keys-certs.def"
 #define XMLDSIGVERIFY_KEY_AND_CERTS_FOLDER              "/var/www/cgi-bin/keys-certs"
@@ -36,7 +37,7 @@ int verify_request(xmlSecKeysMngrPtr mngr);
 int url_decode(char *buf, size_t size);
 
 int
-main(int , char **) {
+main() {
     xmlSecKeysMngrPtr mngr;
 #ifndef XMLSEC_NO_XSLT
     xsltSecurityPrefsPtr xsltSecPrefs = NULL;
-- 
2.39.1

